Sources/Decoder.swift

```diff
@@ -157,6 +157,9 @@ public struct Decoder {
                 
                 for subJSON in jsonArray {
                     guard let model = T(json: subJSON) else {
+                        if T.self is ArrayPartiallyDecodable.Type {
```

Comment:
@hkellaway 🙄  because those are the only one acepting Generics "T", this means you can pass custom types not concrete types like all the other functions. If you still don't understand me the only way to clarify this is for you to try to make this function partially decodable:

`public static func decode(dateArrayForKey key: String, dateFormatter: DateFormatter, keyPathDelimiter: String = GlossKeyPathDelimiter) -> (JSON) -> [Date]? `

You can't apply the same logic here, if you know how please let me know I'm black!